### PR TITLE
refactor: lex + parse integrate test

### LINKED ISSUE #304: [parser] refactoring parser integrate tcs
**Detail**

1. separate different type of stmt will have separate tcs
2. applying scope to integral tcs

### PR DESCRIPTION
**Pull request by zeroFruit**
resolved: #304 

details:

* separate different type of stmt will have separate tcs
* applying scope to integral tcs

- [x] Test case

### CODE REVIEW

File: parse/parser_test.go [hunk 1]
```diff
@@ -17,500 +17,405 @@
...
 
-func testAssignStatementFunc(t *testing.T, fn *ast.FunctionLiteral, tt parserTestCase) {
-	t.Log("test assign statement")
+// TODO: add test cases
+func TestAssignStatement(t *testing.T) {
+	tests := []struct {
+		contractTmpl      contractTmplData
+		expectedFnHeaders []expectedFnHeader
+		expected          []ast.AssignStatement
+	}{}
```
Comment (boohyunsik) [inline parse/parser_test.go]: not add test case yet?

**Review by boohyunsik — APPROVED** [review]
LGTM! but it is better adding some comments for understanding tc's input.
like,
```go
func TestAssignStatement(t *testing.T) {
	tests := []struct {
		contractTmpl      contractTmplData
		expectedFnHeaders []expectedFnHeader
		expected          []ast.AssignStatement
	}{
                // int a = 1 <<
                ...
```

File: parse/parser_test.go (hunk 1 above)
Comment (zeroFruit) [reply parse/parser_test.go]: sorry for confusing you, this pr still WIP.  I want to show you the structure of test cases for feedback

File: parse/parser_test.go [hunk 2]
```diff
@@ -107,13 +107,18 @@ func TestReturnStatement(t *testing.T) {
 		expectedFnHeaders []expectedFnHeader
 		expected          []ast.ReturnStatement
 	}{
+		/*
```
Comment (boohyunsik) [inline parse/parser_test.go:110]: Good 👍

**Comment by zeroFruit** [comment]
I would squash the `test: ` commits into one

**Comment by zeroFruit** [comment]
after #315, #316 merged, this should be merged

**Pull request merged**